src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -0,0 +1,239 @@
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.workspace import ImportFormat
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class AWSRole:
+    path: str
+    role_name: str
+    role_id: str
+    arn: str
+
+
+@dataclass
+class AWSPolicyAction:
+    resource_type: str
+    action: set[str]
+    resource_path: str
+
+
+@dataclass
+class AWSInstanceProfileAction:
+    instance_profile_arn: str
+    resource_type: str
+    actions: str
+    resource_path: str
+    iam_role_arn: str | None = None
+
+
+@dataclass
+class AWSInstanceProfile:
+    instance_profile_arn: str
+    iam_role_arn: str | None = None
+
+    ROLE_NAME_REGEX = r"arn:aws:iam::[0-9]+:(?:instance-profile|role)\/([a-zA-Z0-9+=,.@_-]*)$"
+
+    @property
+    def role_name(self):
+        if self.iam_role_arn:
+            arn = self.iam_role_arn
+        else:
+            arn = self.instance_profile_arn
+        role_match = re.match(self.ROLE_NAME_REGEX, arn)
+        if not role_match:
+            logger.error(f"Role ARN is mismatched {self.iam_role_arn}")
+            return None
+        else:
+            return role_match.group(1)
+
+
+@lru_cache(maxsize=1024)
+def run_command(command):
+    logger.info(f"Invoking Command {command}")
+    process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)
+    output, error = process.communicate()
+    return process.returncode, output.decode("utf-8"), error.decode("utf-8")
+
+
+class AWSResources:
+    S3_ACTIONS: typing.ClassVar[set[str]] = {"s3:PutObject", "s3:GetObject", "s3:DeleteObject", "s3:PutObjectAcl"}
+    S3_REGEX: typing.ClassVar[str] = r"arn:aws:s3:::([a-zA-Z0-9+=,.@_-]*)\/\*"
+
+    def __init__(self, profile: str, command_runner: Callable[[str], tuple[int, str, str]] = run_command):
+        self._profile = profile
+        self._command_runner = command_runner
+
+    def validate_connection(self):
+        validate_command = f"aws sts get-caller-identity --profile {self._profile}"
+        result = self._run_json_command(validate_command)
+        if result:
+            logger.info(result)
+            return True
+        return False
+
+    def list_role_policies(self, role_name: str):
+        list_policies_cmd = (
+            f"aws iam list-role-policies --profile {self._profile} --role-name {role_name} --no-paginate"
+        )
+        policies = self._run_json_command(list_policies_cmd)
+        if not policies:
+            return []
+        yield from policies.get("PolicyNames", [])
+
+    def list_attached_policies_in_role(self, role_name: str):
+        list_attached_policies_cmd = (
+            f"aws iam list-attached-role-policies --profile {self._profile} --role-name {role_name} --no-paginate"
+        )
+        policies = self._run_json_command(list_attached_policies_cmd)
+        if not policies:
+            return []
+        for policy in policies.get("AttachedPolicies", []):
+            yield policy.get("PolicyArn")
```

Comment:
Use normal return, this is harder to mock

Reply:
Fixed